--- a/arch/arm64/boot/dts/qcom/ipq5018.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq5018.dtsi
@@ -129,6 +129,34 @@
 		#size-cells = <2>;
 		ranges;
 
+		bt_region: bt@7000000 {
+			no-map;
+			reg = <0x0 0x07000000 0x0 0x58000>;
+		};
+
+		nss_region: nss@40000000 {
+			no-map = <>;
+			reg = <0x0 0x40000000 0x0 0x01000000>;	/* change to 8MB - 0x00800000 for 256 MB memory profile */
+		};
+
+		uboot@4a800000 {
+			reg = <0x00 0x4a800000 0x00 0x200000>;
+			no-map;
+		};
+
+		sbl@4aa00000 {
+			no-map = <>;
+			reg = <0x00 0x4aa00000 0x00 0x100000>;
+		};
+
+		smem@4ab00000 {
+			compatible = "qcom,smem";
+			reg = <0x0 0x4ab00000 0x0 0x100000>;
+			no-map;
+
+			hwlocks = <&tcsr_mutex 3>;
+		};
+
 		tz_region: tz@4ac00000 {
 			reg = <0x0 0x4ac00000 0x0 0x200000>;
 			no-map;
